File: flatseeker-backend/src/main/java/com/yakzhanov/flatseeker/model/ApartmentRecord.java
package com.yakzhanov.flatseeker.model;

import java.util.Date;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import com.yakzhanov.flatseeker.model.dict.AnimalStatus;
import com.yakzhanov.flatseeker.model.dict.ApartmentType;
import com.yakzhanov.flatseeker.model.dict.BathroomStatus;
import com.yakzhanov.flatseeker.model.dict.LocationStatus;
import com.yakzhanov.flatseeker.model.dict.ProcessStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ToString(onlyExplicitlyIncluded = true)
public class ApartmentRecord {

    @Id
    private String id;

    @ToString.Include
    private String title;
    private String description;

    private Integer rentPrice;
    private Integer feePrice;

    private Integer deposit;

    private Integer area;

    private Boolean conditioner;

    @OneToOne(fetch = FetchType.EAGER)
    private AnimalStatus animalStatus;

    @OneToOne(fetch = FetchType.EAGER)
    private BathroomStatus bathroomStatus;

    private String location;

    @OneToOne(fetch = FetchType.EAGER)
    private ApartmentType apartmentType;

    private Date createdAt;

    private String mainImageUrl;

    // Service
    @OneToOne(fetch = FetchType.EAGER)
    private LocationStatus locationStatus;

    private String link;

    private String platformName;

    @OneToOne(fetch = FetchType.EAGER)
    private ProcessStatus processStatus;

    private Date insertedAt;

    private boolean viewed;

    private boolean removed;

}

File: flatseeker-ui/src/app/model/ApartmentRecord.ts
import { AnimalStatus } from "./dict/AnimalStatus";
import { BathroomStatus } from "./dict/BathroomStatus";
import { ApartmentType } from "./dict/ApartmentType";
import { LocationStatus } from "./dict/LocationStatus";
import { ProcessStatus } from "./dict/ProcessStatus";

export class ApartmentRecord {
  public id: string;
  public title: string;
  public description: string;
  public rentPrice: number;
  public feePrice: number;
  public deposit: number;
  public area: number;
  public conditioner: Boolean;
  public animalsStatus: AnimalStatus;
  public bathroomStatus: BathroomStatus;
  public location: string;
  public apartmentType: ApartmentType;
  public createdAt: Date;
  public mainImageUrl: string;
  public locationStatus: LocationStatus;
  public link: string;
  public platformName: string;
  public processStatus: ProcessStatus;
  public viewed: boolean;

  constructor(id: string, title: string, description: string, rentPrice: number, feePrice: number, deposit: number, area: number, conditioner: Boolean, animalStatus: AnimalStatus, bathroomStatus: BathroomStatus, location: string, apartmentType: ApartmentType, createdAt: Date, mainImageUrl: string, locationStatus: LocationStatus, link: string, platformName: string, processStatus: ProcessStatus, viewed: boolean) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.rentPrice = rentPrice;
    this.feePrice = feePrice;
    this.deposit = deposit;
    this.area = area;
    this.conditioner = conditioner;
    this.animalsStatus = animalStatus;
    this.bathroomStatus = bathroomStatus;
    this.location = location;
    this.apartmentType = apartmentType;
    this.createdAt = createdAt;
    this.mainImageUrl = mainImageUrl;
    this.locationStatus = locationStatus;
    this.link = link;
    this.platformName = platformName;
    this.processStatus = processStatus;
    this.viewed = viewed;
  }

  public static mock() {
    return new ApartmentRecord("", "", "", 0, 0, 0, 0, false,   AnimalStatus.mock(), BathroomStatus.mock(), "", ApartmentType.mock(), new Date(), "", LocationStatus.mock(), "", "", ProcessStatus.mock(), false);
  }
}
